sam-react-app/coreconcept/src/counter/Counter.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {ChangeEvent, Component, createElement} from "react";
+
+type CounterComponentState = {
+    counter: number;
+}
+type CounterComponentProps = {
+    initialValue: string;
+}
+class Counter extends Component<any, CounterComponentState> {
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            counter: Number(props.initialValue) ?? 0,
+        };
+        this.handleInputChange = this.handleInputChange.bind(this);
```

Comment:
There is nothing wrong here but you don't need `binding` if you don't use function declaration for event handlers. You would only need it if you go for `function handleInput(event) { ...}`.

Reply:
Done